projects/plugins/boost/wp-js-data-sync.php

```diff
@@ -259,6 +259,19 @@ function () {
 	}
 );
 
+jetpack_boost_register_readonly_option(
+	'minify_legacy_notice',
+	function () {
+		// If the JETPACK_BOOST_DISABLE_404_TESTER is set and true, we don't need to show the legacy notice.
+		if ( defined( 'JETPACK_BOOST_DISABLE_404_TESTER' ) && JETPACK_BOOST_DISABLE_404_TESTER ) {
+			return false;
+		}
+
+		// Otherwise show it if the 404 tester determined it can't be used.
+		return ! (bool) get_site_option( 'jetpack_boost_static_minification' );
```

Comment:
The flashing notice issue mentioned above is probably because of this. Before the 404 tester has a chance to run, the default value is false as the option doesn't exist yet. 

As a result, when you enable a module, initially the notice will show. After it has a chance to fetch, it hides the notice again. I think something like the following might work:

```suggestion
		/*
		 * Until the 404 tester running, the option will not be created and it will return false.
		 * However, we don't want to show the notice before the tester has a chance to run.
		 * We only want to show the notice when the tester has already run and determined that
		 * it will use the fallback system.
		 */
		return get_site_option( 'jetpack_boost_static_minification' ) !== '';
```

Reply:
Nice catch, I've updated the PR with something similar to your solution.